integration/proxy_tunnel_strategy_test.go

```diff
@@ -0,0 +1,529 @@
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = true
+
+	require.NoError(t, node.CreateEx(t, nil, conf))
+	require.NoError(t, node.Start())
+
+	p.node = node
+}
+
+// makeDatabase bootstraps a new teleport db instance.
+// It connects to a proxy via a reverse tunnel going through a load balancer.
+func (p *proxyTunnelStrategy) makeDatabase(t *testing.T) {
+	if p.db != nil {
+		require.Fail(t, "database already initialized")
+	}
+
+	dbAddr := net.JoinHostPort(Host, strconv.Itoa(ports.PopInt()))
+
+	// setup database service
+	db := NewInstance(InstanceConfig{
+		ClusterName: p.cluster,
+		HostID:      uuid.New().String(),
+		NodeName:    Loopback,
+		log:         utils.NewLoggerForTests(),
+	})
+
+	conf := service.MakeDefaultConfig()
+	conf.AuthServers = append(conf.AuthServers, utils.FromAddr(p.lb.Addr()))
+	conf.Token = "token"
+	conf.DataDir = t.TempDir()
+
+	conf.Auth.Enabled = false
+	conf.Proxy.Enabled = false
+	conf.SSH.Enabled = false
+	conf.Databases.Enabled = true
+	conf.Databases.Databases = []service.Database{
+		{
+			Name:     p.cluster + "-postgres",
+			Protocol: defaults.ProtocolPostgres,
+			URI:      dbAddr,
+		},
+	}
+
+	_, role, err := auth.CreateUserAndRole(p.auth.Process.GetAuthServer(), p.username, nil)
+	require.NoError(t, err)
+
+	role.SetDatabaseUsers(types.Allow, []string{types.Wildcard})
+	role.SetDatabaseNames(types.Allow, []string{types.Wildcard})
+	err = p.auth.Process.GetAuthServer().UpsertRole(context.Background(), role)
+	require.NoError(t, err)
+
+	// start the process and block until specified events are received.
+	require.NoError(t, db.CreateEx(t, nil, conf))
+
+	receivedEvents, err := startAndWait(db.Process, []string{
+		service.DatabasesIdentityEvent,
+		service.DatabasesReady,
+		service.TeleportReadyEvent,
+	})
+	require.NoError(t, err)
+
+	var client *auth.Client
+	for _, event := range receivedEvents {
+		if event.Name == service.DatabasesIdentityEvent {
+			conn, ok := (event.Payload).(*service.Connector)
+			require.True(t, ok)
+			client = conn.Client
+			break
+		}
+	}
+	require.NotNil(t, client)
+
+	// setup a test postgres database
+	postgresDB, err := postgres.NewTestServer(common.TestServerConfig{
+		AuthClient: client,
+		Name:       p.cluster + "-postgres",
+		Address:    dbAddr,
+	})
+	require.NoError(t, err)
+	go postgresDB.Serve()
+
+	p.db = db
+	p.dbAuthClient = client
+	p.postgresDB = postgresDB
+}
+
+// waitForNodeToBeReachable waits for the node to be reachable from all
+// proxies by making sure the proxy peer connectivity info (if any) got
+// propagated to the auth server.
+func (p *proxyTunnelStrategy) waitForNodeToBeReachable(t *testing.T) {
+	check := func(t *TeleInstance, availability int) (bool, error) {
+		nodes, err := t.GetSiteAPI(p.cluster).GetNodes(
+			context.Background(),
+			apidefaults.Namespace,
+		)
+		if err != nil {
+			return false, trace.Wrap(err)
+		}
+
+		for _, nodeT := range nodes {
+			node, ok := nodeT.(*types.ServerV2)
```

Comment:
Can we access (the generic equivalent of) `.Metadata.Name` and `.Spec.ProxyIDs` through `types.Server` instead? Just so we have less changes to chase if we ever changed the concrete type to something else in the future.